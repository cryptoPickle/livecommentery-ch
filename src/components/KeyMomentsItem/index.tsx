import React from "react"
import {View, Image, Text, StyleSheet, TouchableOpacity} from "react-native"
import {largeGap, mediumGap, smallGap} from "../../constants";

export interface KmomentsData {
  id: string
  keyMoment: string,
  avatar: string
  callback: () => void
}


const KeyMomentsItem = (props: KmomentsData) => {
  const {keyMoment, avatar, callback} = props;
  return (
    <TouchableOpacity style={styles.items} onPress={callback} >
        <View style={styles.avatarContainer}>
          <Image source={{uri: avatar}} style={styles.avatar}/>
        </View>
          <Text style={styles.fontMoment}>{keyMoment}</Text>
    </TouchableOpacity>
  )
};

const styles = StyleSheet.create({
  items: {
    flexDirection: "row",
    marginBottom: mediumGap,
    alignItems: "center",
  },
  fontMoment: {
    fontSize: mediumGap,
    color: "#9E9E9E"
  },
  avatar: {
    width: 45,
    height: 45,
  },
  avatarContainer: {
    overflow: "hidden",
    borderRadius: 45,
    width: 45,
    marginRight: largeGap - 10,
  }
});

export default  KeyMomentsItem
