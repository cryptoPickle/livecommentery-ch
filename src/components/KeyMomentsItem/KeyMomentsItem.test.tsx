import React from "react";
import {shallow, ShallowWrapper} from "enzyme";
import KeyMomentsItem from ".";
import {Text, Image, TouchableOpacity, View} from "react-native";




describe("KeyMomentsItem", () => {
  let wrapper: ShallowWrapper;
  let onPressEvent = jest.fn();
  beforeEach(() => {
    wrapper = shallow(<KeyMomentsItem
      avatar="https://api.adorable.io/avatars/285/2"
      keyMoment="Key Moment"
      id="1"
      callback={onPressEvent}
    />)});
  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should have View, Image, TouchableOpacity and Text, ", () => {
    expect(wrapper.find(View)).toHaveLength(1);
    expect(wrapper.find(Image)).toHaveLength(1);
    expect(wrapper.find(TouchableOpacity)).toHaveLength(1);
    expect(wrapper.find(Text)).toHaveLength(1);

  });
  it("should match Avatar link", () => {
    expect(wrapper.find(Image).props().source).toEqual( { uri: 'https://api.adorable.io/avatars/285/2' })
  });
  it("should match Key Moment", () => {
    const comp: any = wrapper;
    expect(comp.find(Text).props().children).toEqual("Key Moment")
  });
  it("should call callback once", () => {
    const comp: any = wrapper;
    comp.find(TouchableOpacity).props().onPress();
    expect(onPressEvent.mock.calls.length).toEqual(1)

  })
});
