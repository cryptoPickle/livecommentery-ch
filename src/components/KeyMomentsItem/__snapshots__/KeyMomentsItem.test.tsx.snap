// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeyMomentsItem should match snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[MockFunction]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "marginBottom": 37.5,
    }
  }
>
  <View
    style={
      Object {
        "borderRadius": 45,
        "marginRight": 65,
        "overflow": "hidden",
        "width": 45,
      }
    }
  >
    <Component
      source={
        Object {
          "uri": "https://api.adorable.io/avatars/285/2",
        }
      }
      style={
        Object {
          "height": 45,
          "width": 45,
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "color": "#9E9E9E",
        "fontSize": 37.5,
      }
    }
  >
    Key Moment
  </Text>
</TouchableOpacity>
`;
